src/lr_apply.rs

```diff
@@ -137,6 +137,7 @@ fn backup4xU<BD: BitDepth>(
     if src_stride < 0 {
         for (src, dst) in src[src_offset - (u - 1) as usize * abs_px_stride..]
             .chunks(abs_px_stride)
+            .take(u as usize)
             .rev()
             .zip(&mut dst[..u as usize])
```

Comment:
Is the reason we need this here because the `.take(u)` needs to be before the `.rev()`?  Because otherwise the `.zip(&mut dst[..u])` should handle it.

Reply:
Yes, `.take(u)` needs to happen before `.rev()`.